tftrt/examples/image-classification/image_classification.py

```diff
@@ -106,28 +122,51 @@ def eval_input_fn():
                 dtype=np.int32)
             labels = tf.identity(tf.constant(labels))
         else:
-            dataset = tf.data.TFRecordDataset(data_files)
-            dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
-            dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
-            dataset = dataset.repeat(count=1)
-            iterator = dataset.make_one_shot_iterator()
-            features, labels = iterator.get_next()
+            if mode == 'validation':
+                dataset = tf.data.TFRecordDataset(data_files)
+                dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
+                dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
+                dataset = dataset.repeat(count=1)
+                iterator = dataset.make_one_shot_iterator()
+                features, labels = iterator.get_next()
+            else:
+                dataset = tf.data.Dataset.from_tensor_slices(data_files)
+                dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
+                dataset = dataset.repeat(count=1)
+                iterator = dataset.make_one_shot_iterator()
+                features = iterator.get_next()
+                labels = np.random.randint(
+                    low=0,
+                    high=get_netdef(model).get_num_classes(),
+                    size=(batch_size),
+                    dtype=np.int32)
+                labels = tf.identity(tf.constant(labels))
         return features, labels
 
     # Evaluate model
+    if mode == 'validation':
+        num_records = get_tfrecords_count(data_files)
+    else:
+        num_records = len(data_files) 
     logger = LoggerHook(
         display_every=display_every,
         batch_size=batch_size,
-        num_records=get_tfrecords_count(data_files))
+        num_records=num_records)
     tf_config = tf.ConfigProto()
     tf_config.gpu_options.allow_growth = True
     estimator = tf.estimator.Estimator(
         model_fn=model_fn,
         config=tf.estimator.RunConfig(session_config=tf_config),
         model_dir='model_dir')
-    results = estimator.evaluate(eval_input_fn, steps=num_iterations, hooks=[logger])
-    
-    # Gather additional results
+    if mode == 'validation':
+        results = estimator.evaluate(input_fn, steps=num_iterations, hooks=[logger])
+    else:
+        results = {}
+        prediction_results = estimator.predict(input_fn, predict_keys=["classes"],  hooks=[logger])
```

Comment:
We would need to use `DurationHook` in `estimator.predict` (but not in estimator.evaluate)`.